app/scripts/views/tour/tour-page-controller.js

```diff
@@ -42,6 +47,15 @@
 
             ctl.onPrevClick = onPrevClick;
             ctl.onNextClick = onNextClick;
+
+            Visualization.get(Config.ids.tour).then(onVisReady, onVisError);
```

Comment:
Hmmm, that may work. The root problem here though is that we need to create the visualization once (in the parent TourController), but then access it in child views multiple times. So either way, get needs to be a promise, because we don't know when we call get in the child controller whether the vis is fully initialized yet or not. And if I moved the callbacks to add, its superfluous because in the current implementation I can just create callbacks on the return value from add anyways (since its the cartodb vis object). 

Not sure what you're referring to when you say 'throwing an error for an expected condition'? 

Hope this clarifies things? I actually think it the current state its a relatively consistent API. The reason this is exists is so the visualization object can be passed around via a service rather than attaching it to the TourController and then accessing it implicitly in the child view, which is far more brittle.

Either way, I'll play with this a bit yet and see if I can come up with something a bit simpler.